### PR TITLE
1.6 Migration

### PR DESCRIPTION
**Pull request by jdoliner**

### CODE REVIEW

File: src/server/admin/server/api_server.go [hunk 1]
```diff
@@ -0,0 +1,353 @@
...
+}
+
+func (a *apiServer) getPachClient() (*client.APIClient, error) {
+	if a.pachClient == nil {
+		var onceErr error
+		a.pachClientOnce.Do(func() {
+			a.pachClient, onceErr = client.NewFromAddress(a.address)
+		})
+		if onceErr != nil {
+			return nil, onceErr
```
[the remark this answers is not in the document]
Comment (jdoliner) [reply src/server/admin/server/api_server.go]: I think that makes sense.

File: src/server/admin/server/api_server.go [hunk 2]
```diff
@@ -0,0 +1,353 @@
...
+					Description: ri.Description,
+				},
+			}); err != nil {
+				return err
+			}
+			cis, err := pachClient.ListCommit(ri.Repo.Name, "", "", 0)
+			if err != nil {
+				return err
+			}
+			for _, ci := range sortCommitInfos(cis) {
```
[the remark this answers is not in the document]
Comment (jdoliner) [reply src/server/admin/server/api_server.go]: I didn't actually wind up going through with allowing `BuildCommit` with an explicit ID to create commits without parents, it messed too many things up including the way we keep track of the size of repos. So now we have to order them.

File: src/client/pfs/pfs.proto [hunk 3]
```diff
@@ -162,6 +162,9 @@ message BuildCommitRequest {
   string branch = 4;
   repeated Commit provenance = 2;
   Object tree = 3;
+  // ID sets the ID of the created commit. It also allows you to BuildCommits
+  // with parents that don't exist.
```
Comment (msteffen) [inline src/client/pfs/pfs.proto]: This comment should be updated

File: src/client/admin/admin.proto [hunk 4]
```diff
@@ -0,0 +1,44 @@
+syntax = "proto3";
+package admin;
+
+import "google/protobuf/empty.proto";
+
+import "client/pfs/pfs.proto";
+import "client/pps/pps.proto";
+
+message Op17 {
```
Comment (msteffen) [inline src/client/admin/admin.proto]: What do you think of using underscore to disambiguate major/minor version (i.e. `Op1_7`)? I was going to confess that it was unconventional, but I think the go linter has an explicit exception for this kind of thing: https://github.com/golang/lint/issues/47 (which, despite the skeptical comment, is fixed by https://github.com/zimmski/lint/commit/2481c997ffa02332f21d9cac2e5e2b3907a51be0)

**Pull request merged**